### PR TITLE
#170258130 Fix bugs

### PR DESCRIPTION
**Pull request by hezronkimutai**
#### What does this PR do?
- Fixes bugs
#### Description of Task to be completed?
- Added a route for fetching a single accommodation
- Added a route for fetching all accommodations
- Modified resend email route to accept email instead of id
- Fixed the get all comment functionality such that all comments can be seen.
- Fixed get a user and create bookings functionalities.
#### How should this be manually tested?
- Clone this repository using the command: `git clone https://github.com/andela/team-odd-bn-backend.git`
- Checkout to this branch using the command: `git checkout bg-manager-email-170258130`
- Install the dependencies by running the command: `npm i`
- Run the application using the command: `npm run dev-start`
- Run the tests using the command: `npm test`
#### Any background context you want to provide?
The name of the branch might be confusing, the reason for this is at the beginning of this task, the plan was to only solve the functionality that sends email to the manager after a successful creation of a trip request. The plan later changed as I and my team mates realised that we have many bugs that could be fixed in only one PR 
#### What are the relevant pivotal tracker stories?
[#170258130](https://www.pivotaltracker.com/n/projects/2407428/stories/170258130)
#### Screenshots (if appropriate)
Screen Shot 2019-12-23 at 12 15 09
Screen Shot 2019-12-23 at 12 15 27
Screen Shot 2019-12-23 at 12 16 19

#### Questions:

### CODE REVIEW

File: src/controllers/AccommodationController.js [hunk 1]
```diff
@@ -71,6 +71,25 @@ class AccommodationController {
...
+   * @param {object} res response object
+   * @memberof AccommodationController
+   * @returns {object} data
+   */
+  static async getAccommodations(req, res) {
+    try {
+      const accommodations = await getAccommodations(req);
+      return Response.successMessage(req, res, 'Accommodation/s successfully fetched', accommodations, 200);
+    } catch (err) {
+      console.log(err);
```
Comment (NiyoEric) [inline src/controllers/AccommodationController.js]: Remove console.

File: src/controllers/BookingController.js [hunk 2]
```diff
@@ -42,6 +42,8 @@ class BookingController {
       return result.length !== 0 ? Response.successMessage(req, res, 'User booking requests are retrieved successfully', result, 200)
         : Response.errorMessage(req, res, 'No Booking request found', 404);
     } catch (error) {
+      console.log(error);
```
Comment (NiyoEric) [inline src/controllers/BookingController.js]: Remove console.

File: src/helpers/EmailHelper.js [hunk 3]
```diff
@@ -31,11 +31,14 @@ class EmailHelper {
...
-   * @param {Object} email the template to use
    * @returns {Object} sendEmail
    */
-  static approveEmailHelper(req, email) {
-    transporter.sendMail(EmailTemplates.approveEmailTemplate(req, email));
+  static approveEmailHelper(req) {
+    try {
+      transporter.sendMail(EmailTemplates.approveEmailTemplate(req));
+    } catch (err) {
+      console.log(err);
```
Comment (NiyoEric) [inline src/helpers/EmailHelper.js]: Remove consoles everywhere.

File: src/helpers/TripHelper.js [hunk 4]
```diff
@@ -87,6 +103,15 @@ class TripHelper {
...
     const maxNumber = Math.max(...countHolder);
+    // SORTS CITIES FROM THE MOST TRAVELLED TO LEAST TRAVELLED
+    // newArray.sort((a, b) => b.counter - a.counter);
+    // console.log(newArray);
+    // const uniqueArray = newArray.filter((thing, index) => {
+    //   const thisThing = JSON.stringify(thing);
+    //   return index === newArray.findIndex(obj => JSON.stringify(obj) === thisThing);
+    // });
+    // console.log('===========', uniqueArray);
+
```
Comment (NiyoEric) [inline src/helpers/TripHelper.js]: Remove comment.

Comment (NiyoEric) [inline src/helpers/TripHelper.js]: Remove comment.

File: src/public/index.html [hunk 5]
```diff
@@ -29,6 +29,7 @@
   socket.on('trip_request_client', (name) => {
     console.log(JSON.parse(name));
   });
- 
+
 </script>
-</html>
+
+</html>
```
Comment (NiyoEric) [inline src/public/index.html]: EOL

File: src/routes/api/accommodationRoute.js [hunk 6]
```diff
@@ -268,5 +268,82 @@ accommodationRoute
...
+ *          description: Unauthorized
+ *        "403":
+ *          description: Forbiden
+ *
+ */
+accommodationRoute
+  .get(
+    '/',
+    verifyToken,
+    // isUserVerified,
```
Comment (NiyoEric) [inline src/routes/api/accommodationRoute.js]: I think this middleware is helpful. what do you think if it is added?

File: src/helpers/EmailHelper.js (hunk 3 above)
Comment (NiyoEric) [inline src/helpers/EmailHelper.js]: Remove console.

**Review by NiyoEric — CHANGES_REQUESTED** [review]
Thanks for getting this done @hezronkimutai, @victorkarangwa4  and @william000000. I think we are good to go. I have left feedback I think it was forgotten.

**Review by NiyoEric — APPROVED** [review]

File: src/controllers/AccommodationController.js [hunk 7]
```diff
@@ -71,6 +71,23 @@ class AccommodationController {
...
+   * @static
+   * @param {object} req  request object
+   * @param {object} res response object
+   * @memberof AccommodationController
+   * @returns {object} data
+   */
+  static async getAccommodations(req, res) {
+    try {
+      const accommodations = await getAccommodations(req);
+      return Response.successMessage(req, res, 'Accommodation/s successfully fetched', accommodations, 200);
```
Comment (victkarangwa) [inline src/controllers/AccommodationController.js]: I think if you want to specify accommodation/accommodations, simply write `accommodation(s)`

**Review by ivymwende — APPROVED** [review]
Looks good to me! Great work Hezron!

**Review by victkarangwa — APPROVED** [review]
Good job!

File: src/database/models/booking.js [hunk 8]
```diff
@@ -3,7 +3,6 @@ module.exports = (sequelize, DataTypes) => {
   const booking = sequelize.define('booking', {
     tripId: DataTypes.INTEGER,
     roomId: DataTypes.INTEGER,
-    userId: DataTypes.INTEGER,
     checkInDate: DataTypes.DATE,
     checkOutDate: DataTypes.DATE
   }, {});
```
Comment (william000000) [inline src/database/models/booking.js:8]: @hezronkimutai  Why have you removed `userId` in this table?

File: src/public/index.html [hunk 9]
```diff
@@ -13,7 +13,7 @@
 </body>
 <script src="/socket.io/socket.io.js"></script>
 <script>
-  const socket = io('http://localhost:3000');
+  const socket = io('https://team-odd-bn-backend-staging.herokuapp.com/');
```
Comment (william000000) [inline src/public/index.html:16]: Nice work @hezronkimutai

File: src/services/BookingService.js [hunk 10]
```diff
@@ -102,12 +107,12 @@ class BookingService {
...
-      where: { userId }, raw: true
-    };
-
-    const result = !userId ? await CommonQueries.findAll(booking) : await CommonQueries.findAll(booking, findOneBookingObj);
+    const { bookingId: id } = req.params;
+    const findOneBookingObj = id
+      ? {
+        where: { id }, raw: true
+      } : { raw: true };
+    const result = await CommonQueries.findAll(booking, findOneBookingObj);
```
Comment (william000000) [inline src/services/BookingService.js]: @william000000 :  may I know your intention here?
@hezzie : My intention is to fetch user bookings by using joint table queries, As I had told you earlier, I removed the userId column from the booking table so what what we are getting in this function is an object with user bookings and all the bookings of the users under one manager, in this case if the user is not a manager, we only get an empty array.

**Review by william000000 — COMMENTED** [review]
You have made a good initiative @hezronkimutai, However, there are some changes you made that you may explain to me well.

**Review by shemaeric — APPROVED** [review]
LGTM

**Pull request merged**